server/storage/schema/schema.go

```diff
@@ -0,0 +1,135 @@
+// Copyright 2021 The etcd Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package schema
+
+import (
+	"fmt"
+
+	"github.com/coreos/go-semver/semver"
+	"go.uber.org/zap"
+
+	"go.etcd.io/etcd/server/v3/storage/backend"
+)
+
+var (
+	V3_5 = semver.Version{Major: 3, Minor: 5}
+	V3_6 = semver.Version{Major: 3, Minor: 6}
+)
+
+// Migrate updates storage version to provided target version.
+func Migrate(lg *zap.Logger, tx backend.BatchTx, target semver.Version) error {
+	ver, err := detectStorageVersion(lg, tx)
+	if err != nil {
+		return fmt.Errorf("cannot determine storage version: %w", err)
+	}
+	if ver.Major != target.Major {
+		lg.Panic("Chaning major storage version is not supported",
+			zap.String("storage-version", ver.String()),
+			zap.String("target-storage-version", target.String()),
+		)
+	}
+	if ver.Minor > target.Minor {
+		lg.Panic("Downgrades are not yet supported",
```

Comment:
Should we panic or return error maybe instead?

Maybe: 

```suggestion
		lg.Panic("Target version is lower than the current version, specific higher version, downgrades are not yet supported",
```

Reply:
This is one breaking change we are introducing, future etcd version (for example v3.7) cannot be passed to v3.6 without doing the downgrade process.